### PR TITLE
Normative: Add Intl.Segmenter

### PR DESCRIPTION
**Pull request by gibson042**
* [ ] @leobalter

### CODE REVIEW

**Comment by FrankYFTang** [comment]
Should the "2 Conformance" https://tc39.es/ecma402/#conformance also be changed for Intl.Segmenter?

**Comment by FrankYFTang** [comment]
I suggest you also add 
```
    <li>
      <a href="https://www.unicode.org/reports/tr29/">Unicode Standard Annex 29- Unicode Text Segmentation</a>
    </li>  
``` 
to the end of normative-references.html
And also mayb consider listing 
Common Locale Data Repository (available at http://cldr.unicode.org)
and
International Components for Unicode https://unicode-org.github.io/icu-docs/
Since that two are also mentioned in the proposal.

**Comment by gibson042** [comment]
Thank you @FrankYFTang. I have included UAX 29, but not CLDR or ICU because references to those are intentionally non-normative.

**Comment by gibson042** [comment]
@ryzokuken Could I get a review from you as well?

**Comment by ryzokuken** [comment]
@gibson042 absolutely! Sorry for the delay, but I thought this would be an even larger diff. On it.

**Review by ryzokuken — APPROVED** [review]
Overall LGTM, with a few comments inline. Two main comments:

1. Would you like to switch to the new structured arguments syntax?
2. The AOs are interleaved all over the spec and I don't see a compelling reason for that. Would you mind putting the AO section at the top with all the AOs in it?

File: spec/segmenter.html
Comment (gibson042) [reply spec/segmenter.html:81]: I think it is better than hardcoding, because https://tc39.es/ecma402/#sec-internal-slots documents a [[RelevantExtensionKeys]] for each service constructor.

File: spec/segmenter.html
Comment (gibson042) [reply spec/segmenter.html:200]: I think there's a lot of value in positioning Create\<Type> AOs as the first subsection of their \<Type> section and following them with the description of the prototype they use, supporting a linear reading of the specification.

File: spec/segmenter.html
Comment (gibson042) [reply spec/segmenter.html:222]: `containing` is not a method of _Segmenter_ instances (the values returned from `new Intl.Segmenter(…)`, whose prototype is %Segmenter.prototype%, which can be accessed as the initial value of Intl.Segmenter.prototype), but rather of _Segments_ instances (the values returned from `segmenter.segment(str)`, whose prototype is the different object %SegmentsPrototype%, for which there is no simple access path available to ECMAScript code).

File: spec/segmenter.html
Comment (gibson042) [reply spec/segmenter.html:389]: As with CreateSegmentsObject, I think positioning operations below their use makes for better (specifically more linear) consumption of the spec. I'm willing to change this, but would rather update other sections to follow its pattern. Thoughts?

**Review by gibson042 — COMMENTED** [review]
> Would you like to switch to the new structured arguments syntax?

Yes, but I think I'd prefer that change after merge, to minimize divergence from the text of the proposal (which preceded structured headers).

> The AOs are interleaved all over the spec and I don't see a compelling reason for that. Would you mind putting the AO section at the top with all the AOs in it?

How strongly do you feel about this? I have a fairly strong preference for this pattern (with the compelling reason being "better readability for linear spec consumption"), and would like to update the rest of ECMA-402 accordingly as soon as possible. But I'm willing to revert this to the existing pattern if you think that will be contentious or take too long.

File: spec/segmenter.html
Comment (ryzokuken) [reply spec/segmenter.html:21]: SGTM, thanks.

File: spec/segmenter.html
Comment (ryzokuken) [reply spec/segmenter.html:77]: Makes sense.

File: spec/segmenter.html
Comment (ryzokuken) [reply spec/segmenter.html:221]: sorry, I mistyped, I meant `Segments.prototype.containing`, didn't know that this wasn't valid.

**Comment by ryzokuken** [comment]
@gibson042 

> How strongly do you feel about this? I have a fairly strong preference for this pattern (with the compelling reason being "better readability for linear spec consumption"), and would like to update the rest of ECMA-402 accordingly as soon as possible. But I'm willing to revert this to the existing pattern if you think that will be contentious or take too long.

Honestly, I'm not sure anymore. Your rationale sounds great to me, but there's also the benefit of keeping things organized in a single location, maybe this is especially notable here because the proposal introduces three types simultaneously. Perhaps the better judge of this would be the implementers?

@FrankYFTang @Constellation does this new pattern of organizing AOs seem helpful to you? Do you think it correctly corresponds to how you consume the spec? If so, I have no reservations against it.

@gibson042 should we also solicit feedback from the 262 editors here?

**Comment by gibson042** [comment]
> should we also solicit feedback from the 262 editors here?

Possibly, but 262 itself is inconsistent (e.g., [Map](https://tc39.es/ecma262/multipage/keyed-collections.html#sec-map-objects) follows the pattern I'm proposing here but [ArrayBuffer](https://tc39.es/ecma262/multipage/structured-data.html#sec-arraybuffer-objects) is more like the rest of 402, and [WeakRef](https://tc39.es/ecma262/multipage/managing-memory.html#sec-weak-ref-objects) is in between them).

**Comment by sffc** [comment]
2022-01-13 discussion: https://github.com/tc39/ecma402/blob/master/meetings/notes-2022-01-13.md#normative-add-intlsegmenter-553

File: spec/segmenter.html
Comment (gibson042) [reply spec/segmenter.html:21]: Captured at #649

**Comment by gibson042** [comment]
@ryzokuken #650

**Pull request merged**
